src/rest/connector/libs/nd/implementation.py

```diff
@@ -297,36 +291,86 @@ def post(self, api_url, payload, expected_status_code=requests.codes.ok,
-                                   "expected status code '{e}'"
-                                   ", got:\n {msg}".format(furl=full_url,
-                                                  d=self.device.name,
-                                                  c=response.status_code,
-                                                  e=expected_status_code,
-                                                  msg=response.text))
+            raise RequestException(f"POST {full_url} to {self.device.name} has returned the "
+                                   f"following code '{response.status_code}', instead of the "
+                                   f"expected status code '{expected_status_code}'"
+                                   f", got:\n {response.text}")
         return output
 
+    @BaseConnection.locked
+    @isconnected
+    def post(self, api_url, params=None, data=None, json=None, files=None,
+             timeout=30, retries=3, retry_wait=10) -> requests.Response:
+        """POST REST Command to configure information from the device
+        Arguments
+        ---------
+            api_url (string): subdirectory part of the API URL
+            params (dict): Query string parameters
+            data (dict):
+            json (json) : if request header Content-Type is application/json
+            files (dict):
+            expected_status_code (int): Expected result
+            timeout (int): Maximum time
+        """
+
+        if not self.connected:
+            raise Exception(f"'{self.device.name}' is not connected for "
+                            f"alias '{self.alias}'")
+        # Eliminate the starting "/" if present, as it may cause problems
+        api_url = api_url.lstrip('/')
+        # Deal with the url
+        full_url = f'{self.url}{api_url}'
+
+        for _ in range(retries):
+            try:
+                if params and not json:
+                    response = self.session.post(full_url, params=params, timeout=timeout,
+                                                verify=self.verify)
+                elif params and json:
+                    response = self.session.post(full_url,params=params, json=json,
+                                                 timeout=timeout, verify=self.verify)
+                elif json and not params:
+                    response = self.session.post(full_url,json=json,
+                                                 timeout=timeout, verify=self.verify)
+                elif data:
+                    response = self.session.post(full_url, data=data,
+                                                 timeout=timeout, verify=self.verify)
+                elif files:
+                    response = self.session.post(full_url, files=files,
+                                                 timeout=timeout, verify=self.verify)
+                else:
+                    response = self.session.post(full_url, timeout=timeout, verify=self.verify)
```

Comment:
Are we not able to pass `None` to the values? IE, surely if `json` is `None` we can still do `self.session.post(..., json=None)`?

Would be better than having this large if/else structure